MetalK8s UI fix the bugs in the dashboard page

ChengYanJin opened the pull request:
**Component**: UI

**Context**: 
We introduced several bugs in dashboard page.

- platform active alerts should filter out the alerts not related to platform
- the tooltip in services is not visible
- the scroll is broken in the three panels 
- when the label is at very beginning, it’s half hidden, same for the yaxis title
- when switching from last 1h=> last 7days the charts are broken 
- remove the "Planes"title 
- Better align the View All 
- the title for the charts is using the wrong font-spacing

Not yet address
- the responsiveness is broken on global health when resizing the window 

**Summary**:
Embed the new core-ui version v0.21.7 to fix the bug of first label in xAxis and add the chart titel

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/charts: Solve the issue that the charts don't get updated

Bug: When we switch timespan, the charts don't get updated even through
we get the new data

ChengYanJin committed:
ui/dashboard: Remove the Plane text in netwrok panel

ChengYanJin committed:
ui/dashboard: Fix the View all text far away from the number

ChengYanJin committed:
ui/dashboard: Fix the scroll issue in the 3 panels

ChengYanJin committed:
ui/package: Bump the core-ui to v0.21.7

ChengYanJin committed:
shell-ui: Bump the core-ui to v0.21.7

ChengYanJin committed:
ui/dashboard: Platform alerts should only contains metalk8s alerts

In order to retrieve all the alerts fired by MetalK8s, we recurrsively
query the alerts thanks to the childrenJsonPath from the top level alert
('ClusterAtRisk', 'ClusterDegraded') to the atomic alerts.

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @chengyanjin

* @nicolast

* @jbwatenbergscality

* @teddyandrieux

* @lucieleonard

JBWatenbergScality commented on ui/src/containers/DashboardPage.js:
```diff
@@ -95,13 +94,18 @@ const DashboardPage = () => {
-        <div className="inventory">
+        <div
+          className="inventory"
+          style={{ height: '100%', overflowY: 'auto' }}
```
Can it be replaced by `<DashboardScrollableArea>` ?

ChengYanJin replied:
Indeed, for sure ;)

JBWatenbergScality commented on ui/src/containers/DashboardPage.js:
```diff
@@ -95,13 +94,18 @@ const DashboardPage = () => {
         </div>
-        <div className="network">
+
+        <div className="network" style={{ height: '100%', overflowY: 'auto' }}>
```
Same here

ChengYanJin replied:
solved in 72b2c58734f5a2d565eae9d2e7f93f3aedda6625

JBWatenbergScality commented on shell-ui/src/alerts/services/alertUtils.js:
```diff
@@ -114,6 +114,7 @@ export const formatActiveAlerts = (alerts: Array<PrometheusAlert>): Alert[] => {
             alert.annotations.selectors.split(',')) ||
           [],
       },
+      childrenJsonPath: alert.annotations.childrenJsonPath || '',
```
Why `|| ''` ? I would better set it to undefined if the Json path is not defined

JBWatenbergScality commented on ui/src/services/alertUtils.js:
```diff
@@ -198,36 +201,61 @@ Format the alerts from Loki.
+            (alert.annotations && alert.annotations.runbook_url) || '',
+          labels: {
+            ...alert.labels,
+            parents:
```
Not really related to this or but parents annotations has been removed

ChengYanJin replied:
solved in 72b2c58734f5a2d565eae9d2e7f93f3aedda6625

ChengYanJin committed:
ui: Fix the scroll issue in node/volume metric tab

ChengYanJin committed:
ui/alerts: Remove the parents field in the formatted Alert

ChengYanJin committed:
ui/alerts: Set undefined for childrenJsonPath if it doesn't exist

ChengYanJin committed:
ui/dashboard: Wrap DashboardScrollableArea for the scrollable panel

ChengYanJin committed:
ui/package: Add jsonpath package

ChengYanJin committed:
ui/tests: Fix the test becase we remove the parents field

ChengYanJin committed:
ui/dashboard: Fix the global health responsiveness issue

ChengYanJin committed:
ui/metrics: Add max-width for the metric chart

To make sure when we resize the screen, we will still able to view the
entire metric chart

ChengYanJin committed:
ui/dashboard: Memorize the metalk8sAtomicAlerts

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.11`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye chengyanjin.
